### PR TITLE
Added wallet support & Changed bonding contract to staking

### PR DESCRIPTION
**Pull request by codeaddict99**
- Added walletconnect wallet support
#19 

- Changed "Bonding Contract" Contract name to "Staking"
#164

### CODE REVIEW

**Comment by Zequez** [comment]
I'm feeling sad; it looks like quite a lot of effort was put into the renaming thing, but simply renaming it in the UI would have sufficed.
I'm also feeling angry because there are 111 files to review, including all the contracts deployments. Contract files themselves have been modified, but their deployed version has not. This comes with compatibility and security issues which I have zero interest in managing.

If it comes to a vote, I vote for the changes to be reduced to just make the renaming on a visual-level, without changing the contracts. What if tomorrow we decide to rename the feature to "Super Ubiquity Fun Trust Fund (SUFTF for short)"; are we going to rename everything all the time?

@pavlovcik what's your take in this?

As for the WalletConnect thing I still gotta check it out.

**Comment by 0x4007** [comment]
If the tests pass for the smart contracts then should be fine. But all that matters for this deliverable are the UI changes as we do not plan to redeploy the smart contracts.

**Comment by codeaddict99** [comment]
I thought that we should have the same naming convention throughout the whole project whether it is UI or smart contracts.
I'll revert the changes in the smart contracts so that we can work in the current situation without conflict.

**Comment by kecrypto** [comment]
Reverted changes in smart contracts.
Now you don't need any renaming.

**Comment by sergfeldman** [comment]
@kecrypto Can you please check that a similar error does not appear in this PR 
https://github.com/ubiquity/ubiquity-dollar-development/pull/195#issuecomment-1175601145

**Comment by codeaddict99** [comment]
> @kecrypto Can you please check that a similar error does not appear in this PR [#195 (comment)](https://github.com/ubiquity/ubiquity-dollar-development/pull/195#issuecomment-1175601145)

I checked it again and no issues.

**Comment by sergfeldman** [comment]
@pavlovcik @Zequez Please finish reviewing this PR

File: components/lib/hooks/contracts/useManagerManaged.tsx [hunk 1]
```diff
@@ -6,9 +6,9 @@ import {
   DollarMintingCalculator__factory,
   UbiquityAutoRedeem__factory,
   UbiquityGovernance__factory,
-  BondingShareV2__factory,
```
Comment (Zequez) [inline components/lib/hooks/contracts/useManagerManaged.tsx:9]: Here is where I would draw the line. Do not modify the smart contracts; just import it as BondingV2 and name it stake if you want. Even thought, as mentioned, just changing the text on the screen would have sufficed.

File: components/monitor/PriceMonitor.tsx [hunk 2]
```diff
@@ -75,6 +75,8 @@ const PriceMonitorContainer = () => {
 };
 
 const PriceMonitor = (props: PriceMonitorProps) => {
+  console.log("price mointor");
```
Comment (Zequez) [inline components/monitor/PriceMonitor.tsx]: ☝️

File: components/launch-party/FundingPools.tsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import BondingPool from "./BondingPool";
```
Comment (Zequez) [inline components/launch-party/FundingPools.tsx:1]: I don't think the original issue was for the launch party Bonding Pools to be renamed to staking. The whole idea was for the other feature to not be confused with this one. So I think all of these on the launch-party should be rollbacked. @pavlovcik what's your take?

File: contracts/dollar/contracts/Bonding.sol [hunk 4]
```diff
@@ -15,281 +15,189 @@ import "./interfaces/IERC1155Ubiquity.sol";
 import "./utils/CollectableDust.sol";
 
 contract Bonding is CollectableDust {
-    using SafeERC20 for IERC20;
```
Comment (Zequez) [inline contracts/dollar/contracts/Bonding.sol:18]: Just don't modify any of the contracts please. I'm seeing 14 .sol files modified.

File: contracts/dollar/package.json [hunk 5]
```diff
@@ -13,18 +13,18 @@
     "url": "https://github.com/ubiquity/uad-contracts.git"
   },
   "scripts": {
-    "clean": "yarn hardhat clean",
```
Comment (Zequez) [inline contracts/dollar/package.json:16]: Was this a merge issue thing? This file shouldn't have been modified.

File: contracts/ubiquistick/package.json [hunk 6]
```diff
@@ -4,15 +4,15 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "clean": "yarn hardhat clean",
```
Comment (Zequez) [inline contracts/ubiquistick/package.json:7]: This file shouldn't have been modified.

**Review by Zequez — CHANGES_REQUESTED** [review]
Aside from the issues resulting from modifying the contracts, it looks like there is a type issue with using the WalletConnectProvider.
Make sure that running `yarn build` actually compiles; Vercel is not deploying.
Screen Shot 2022-07-08 at 04 20 38
Screen Shot 2022-07-08 at 04 20 49

**Comment by 0x4007** [comment]
https://github.com/FleekHQ/fleek-storage-js/issues/25#issuecomment-830119913 @kecrypto

**Comment by codeaddict99** [comment]
Build succeeded!
https://bafybeida4hd3dy7fck7ldskzz4la54fyxnxsd7p7xbj5ag7d6zb7zuqezi.on.fleek.co/

**Comment by Zequez** [comment]
Some things to consider:

- The changes to the contracts are just formatting; but there is no solidity formatting configured; what auto-formatter are you using @kecrypto?  Maybe we could add https://www.npmjs.com/package/prettier-plugin-solidity . Since I don't really want to keep asking you to change stuff I say we can just go ahead and pass it.
- However, the changes to package.json on both contracts repositories and the removal of yarn.lock and replacement with with package-lock.json on the root repo were not in the scope of the tasks, neither was it reported as an intentional change. So please do revert that.
- As for the WalletConnect feature, it looks fine.

**Comment by Zequez** [comment]
@kecrypto  WalletConnect feature looks fine, yes, but on deeper testing, it doesn't work fine. Have you actually tried using a QR to connect a mobile wallet? There is a reference to `process.env.API_KEY_ALCHEMY` on the useWeb3.tsx file that [is not defined on Next.js browser loaded files](https://nextjs.org/docs/basic-features/environment-variables), so I'm not sure how you managed to run it. Either way when actually scanning the QR with the mobile Metamask wallet it doesn't actually work either, even by hardcoding the API_KEY_ALCHEMY value.

How did you test that the feature works?

File: contracts/dollar/package.json (hunk 5 above)
Comment (0x4007) [reply contracts/dollar/package.json:16]: I think this is technically correct, to exclude `yarn`

If it works then it settles the matter.

**Comment by 0x4007** [comment]
@kecrypto https://github.com/kecrypto/ubiquity-dollar-development/pull/1

**Comment by sergfeldman** [comment]
@kecrypto Please, merge the changes from Alex ^ into your PR and make sure everything is correct.
After that, please post a comment that this PR is ready for merging.

**Comment by codeaddict99** [comment]
Fixed all the issues and merged Alex's PR.
Please check again.

**Comment by 0x4007** [comment]
> Fixed all the issues and merged Alex's PR. Please check again.

There's a ton of merge conflicts so there's a ton of issues remaining.

**Comment by sergfeldman** [comment]
@kecrypto Please note that the resolution of these 2 bounties took many times longer than planned.
You have time to fully complete this PR until the end of July 31st, 2022.
If by this time this PR is not merged, these 2 bounties will be terminated for you.

**Pull request closed without merging**